### PR TITLE
Add compiler plugins directive

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

**Comment by romanowski** [comment]
cc @kubukoz

**Review by kubukoz — APPROVED** [review]

File: modules/build/src/main/scala/scala/build/preprocessing/directives/UsingCompilerPluginDirectiveHandler.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import scala.build.preprocessing.ScopePath
+import scala.build.{Position, Positioned}
+
+case object UsingCompilerPluginDirectiveHandler extends UsingDirectiveHandler {
+  def name             = "Compiler plugins"
+  def description      = "Adds compiler plugins"
+  def usage            = "using plugin org:name:ver | using plugins org:name:ver org2:name2:ver2"
+  override def usageMd = "`using plugin `_org_`:`name`:`ver"
+  override def examples = Seq(
+    "using lib org.typelevel:::kind-projector:0.13.2"
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/directives/UsingCompilerPluginDirectiveHandler.scala]:
```suggestion
    "using plugin org.typelevel:::kind-projector:0.13.2"
```

File: modules/build/src/main/scala/scala/build/preprocessing/directives/UsingCompilerPluginDirectiveHandler.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          }
+        )
+      case _ => None
+    }
+
+  private def parseDependency(depStr: String): Either[BuildException, AnyDependency] =
+    DependencyParser.parse(depStr)
+      .left.map(err => new DependencyFormatError(depStr, err))
+
+  override def keys = Seq("lib", "libs")
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/directives/UsingCompilerPluginDirectiveHandler.scala:46]:
```suggestion
  override def keys = Seq("plugin", "plugins")
```

**Review by lwronski — APPROVED** [review]
LGTM overall

**Pull request merged**